costnav_isaacsim/README.md

```diff
@@ -299,6 +312,335 @@ navigator.waitUntilTaskComplete()
+    initial_pose_delay=1.0,        # Delay after initial pose (seconds)
+    goal_delay=0.5,                # Delay after goal pose (seconds)
+    teleport_height=0.1,           # Teleportation height offset (meters)
+    robot_prim_path="/World/Nova_Carter_ROS",  # Robot prim path
+    teleport_settle_steps=5,       # Physics settling steps after teleport
+)
+
+# Create mission manager with custom config
+manager = MissionManager(
+    mission_config=mission_config,
+    simulation_context=simulation_context,
+    manager_config=manager_config,
+)
+
+# Main simulation loop
+while running:
+    simulation_context.step(render=True)
+    manager.step()
+```
+
+### Architecture: State Machine-Based Execution
+
+The MissionManager uses a state machine to ensure proper synchronization between Isaac Sim physics and Nav2:
+
+**Mission States:**
+1. **INIT**: Initialize ROS2 node, NavMesh sampler, and marker publisher
+2. **WAITING_FOR_NAV2**: Wait for Nav2 stack to become ready
+3. **READY**: Sample new start/goal positions from NavMesh
+4. **TELEPORTING**: Teleport robot to start position in Isaac Sim
+5. **SETTLING**: Wait for physics to settle after teleportation (5 steps)
+6. **PUBLISHING_INITIAL_POSE**: Publish initial pose to `/initialpose` for AMCL
+7. **PUBLISHING_GOAL**: Publish goal pose to `/goal_pose` for Nav2
+8. **WAITING_FOR_COMPLETION**: Monitor navigation progress
+9. **COMPLETED**: All missions finished
+
+This state machine approach ensures:
+- Physics simulation steps occur between teleportation and pose publishing
+- Proper timing delays for AMCL initialization
+- Clean separation of concerns for each mission phase
+
+### Features
+
+- **State Machine Execution**: Proper synchronization with simulation loop
+- **NavMesh Position Sampling**: Sample valid start/goal positions from Isaac Sim's NavMesh
+- **Distance Constraints**: Enforce minimum (5m) and maximum (50m) distance between positions
+- **Path Validation**: Verify navigable paths exist between sampled positions
+- **Robot Teleportation**: Teleport robot to start position in Isaac Sim with physics settling
+- **AMCL Initialization**: Publish initial pose for localization
+- **RViz Markers**: Visualize start (green), goal (red), and robot (blue) positions
+- **Integrated Launch**: Run missions alongside simulation with `--mission` flag
+- **Auto-Setup**: Automatically configures Isaac Sim teleportation if robot_prim_path is provided
+
+### RViz Marker Topics
+
+The MissionManager publishes visualization markers for debugging and monitoring:
+
+| Topic          | Type | Color | Description                    |
+| -------------- | ---- | ----- | ------------------------------ |
+| `/start_marker`| `visualization_msgs/Marker` | Green (0, 1, 0) | Start position (ARROW marker)  |
+| `/goal_marker` | `visualization_msgs/Marker` | Red (1, 0, 0)   | Goal position (ARROW marker)   |
+| `/robot_marker`| `visualization_msgs/Marker` | Blue (0, 0, 1)  | Current robot position (10 Hz) |
```

Comment:
I see there should be a robot marker in blue, but green and red marker is only visible

Reply:
I found out that /odom is not being published, which is the topic tracked by the blue marker.